cmake/DaemonFlags.cmake

```diff
@@ -148,13 +148,20 @@ else()
     set(WARNMODE "no-")
 endif()
 
+# Compiler options
+option(USE_FLOAT_EXCEPTIONS "Use floating point exceptions with common.floatException.* cvars" OFF)
 option(USE_FAST_MATH "Use fast math" ON)
 
-# Compiler options
+if (USE_FLOAT_EXCEPTIONS)
+	add_definitions(-DDAEMON_USE_FLOAT_EXCEPTIONS)
+endif()
+
 if (MSVC)
     set_c_cxx_flag("/MP")
 
-	if (USE_FAST_MATH)
+	if (USE_FLOAT_EXCEPTIONS)
+		set_c_cxx_flag("/fp:strict")
```

Comment:
I tried on MSVC now and it works, but apparently the flag enables some warning spam. So I suggest this patch
```diff
--- a/cmake/DaemonFlags.cmake
+++ b/cmake/DaemonFlags.cmake
@@ -161,6 +161,9 @@ if (MSVC)

        if (USE_FLOAT_EXCEPTIONS)
                set_c_cxx_flag("/fp:strict")
+               # Don't switch on C4305 "truncation from 'double' to 'float'" every
+               # time an unsuffixed decimal constant is used
+               set_c_cxx_flag("/wd4305")
        elseif (USE_FAST_MATH)
                set_c_cxx_flag("/fp:fast")
        endif()
```